rust/benches/caching.rs

```diff
@@ -0,0 +1,322 @@
+                ))
+            }
+            CacheType::LockCache => Caches::LockCache(lock_cache::LockCache::new(
+                size as usize,
+                Arc::new(ProbPinnedHook { prob: pinning_prob }),
+            )),
+        }
+    }
+}
+
+impl Caches {
+    /// Fills the cache to its capacity, using ids in range 0..capacity
+    pub fn fill(&mut self) {
+        for i in 0..self.capacity() {
+            match self {
+                Caches::QuickCache(cache) => {
+                    cache.insert(i, 42i64);
+                }
+                Caches::LockCache(lock_cache) => {
+                    let _unused = lock_cache
+                        .get_read_access_or_insert(i, || Ok(42i64))
+                        .unwrap();
+                }
+                Caches::CachedNodeManager(node_manager) => {
+                    let _unused = node_manager.get_read_access(i).unwrap();
+                }
+            }
+        }
+    }
+
+    /// Returns the capacity of the cache
+    pub fn capacity(&self) -> u64 {
+        match self {
+            Caches::QuickCache(cache) => cache.capacity(),
+            Caches::CachedNodeManager(node_manager) => node_manager.capacity(),
+            Caches::LockCache(lock_cache) => lock_cache.capacity(),
+        }
+    }
+
+    /// Executes a read operation on the cache for the given id
+    pub fn execute_read_op(&self, iter: u64) {
+        match self {
+            Caches::QuickCache(cache) => match cache.get_value_or_guard(&iter, None) {
+                quick_cache::sync::GuardResult::Guard(guard) => {
+                    let _ = guard.insert(42i64);
+                }
+                quick_cache::sync::GuardResult::Value(_) => {}
+                quick_cache::sync::GuardResult::Timeout => {
+                    unreachable!()
+                }
+            },
+            Caches::CachedNodeManager(node_manager) => {
+                let _node = node_manager.get_read_access(iter).unwrap();
+            }
+            Caches::LockCache(lock_cache) => {
+                let _node = lock_cache
+                    .get_read_access_or_insert(iter, || Ok(42i64))
+                    .unwrap();
+            }
+        }
+    }
+}
+
+/// Benchmark caches read performance.
+/// It varies:
+/// - Cache size (influence contention)
+/// - Whether the accessed ids are in cache or not (cache hit/miss)
+/// - Number of threads (influence contention)
+fn read_benchmark(c: &mut criterion::Criterion) {
+    fastrand::seed(123);
+
+    let cache_sizes = [100_000, 1_000_000];
+    for cache_size in &cache_sizes {
+        for in_cache in [true, false] {
+            let get_id = |i: u64| {
+                if in_cache {
+                    i % cache_size
+                } else {
+                    i + cache_size
+                }
+            };
+            let mut bench_group =
+                c.benchmark_group(format!("caching/read/Size:{cache_size}/InCache:{in_cache}"));
```

Comment:
I'd like to keep the naming consistent with the benchmarks we already have.
```suggestion
                c.benchmark_group(format!("caching/read/{cache_size}capacity/{in_cache}cached"));
```